internal/buildinfo/version.go

```diff
@@ -2,23 +2,164 @@ package buildinfo
 
 import (
 	"errors"
-	"fmt"
-	"os"
 	"runtime/debug"
-	"strconv"
+	"strings"
 	"time"
 
 	"github.com/blang/semver"
+	"github.com/loadsmart/calver-go/calver"
 )
 
+type Version interface {
+	String() string
+	EQ(other Version) bool
+	Newer() bool
+	SeverelyOutdated(other Version) bool
+}
+
+type SemverVersion struct {
+	semver.Version
+}
+
+func ParseSemver(version string) (*SemverVersion, error) {
+	v, err := semver.ParseTolerant(version)
+	if err != nil {
+		return nil, err
+	}
+	return &SemverVersion{v}, nil
+}
+
+func (v SemverVersion) EQ(other Version) bool {
+	o, ok := other.(*SemverVersion)
+	if ok {
+		return v.Version.EQ(o.Version)
+	} else {
+		return false
+	}
+}
+
+func (v *SemverVersion) Newer() bool {
+	_, ok := parsedVersion.(*CalverVersion)
+	if ok {
+		return false
+	} else {
+		other := parsedVersion.(*SemverVersion)
+		return v.Version.GT(other.Version)
+	}
+}
+
+func (v *SemverVersion) SeverelyOutdated(other Version) bool {
+	otherVer, ok := other.(*SemverVersion)
+	if ok {
+		if v.Major < otherVer.Major {
+			return true
+		}
+		if v.Minor < otherVer.Minor {
+			return true
+		}
+		if v.Patch+5 < otherVer.Patch {
+			return true
+		}
+		return false
+	} else {
+		return false
+	}
+}
+
+type CalverVersion struct {
+	calver.Version
+}
+
+const calverFormat = "YYYY.0M.0D.MICRO"
+
+func ParseCalver(version string) (*CalverVersion, error) {
+	v, err := calver.Parse(calverFormat, version)
+	if err != nil {
+		// Does the version parse as calver without dashes? Needed for semverr
+		// compatibility for goreleaser.
+		dedashedVersion := strings.Replace(version, "-", ".", 1)
+		v, err = calver.Parse(calverFormat, dedashedVersion)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return &CalverVersion{*v}, nil
+}
+
```

Comment:
We could add a String method here that adds the dashes back, so we can keep our formatting consistent.

This *should* work?
```suggestion

func (v CalverVersion) String() string {
	verStr := []rune(v.Version.String())
	count := 0
	for i, v := range verStr {
		if v == '.' {
			count++
		}
		if count == 3 {
			verStr[i] = '-'
			break
		}
	}
	return string(verStr)
}

```